src/main/java/de/metas/ui/web/pickingV2/packageable/process/PackageablesView_PrintPicklist.java

```diff
@@ -106,12 +109,34 @@ protected String doIt()
 		return MSG_OK;
 	}
 
-	private void pickIfNeeded(final PackageableRow row)
+	private void createPickingCandidatesIfNeeded(final PackageableRow row)
 	{
 		final boolean existsPickingCandidates = pickingCandidateService.existsPickingCandidates(row.getShipmentScheduleIds());
 		if (!existsPickingCandidates)
 		{
-			productsToPickRowsRepository.pick(row);
+
+			Services.get(ITrxManager.class).runInNewTrx(new TrxRunnable2()
```

Comment:
doesn't it suffice to use `TrxRunnable` here? if so, you could instead write
```java
runInNewTrx(trx-> productsToPickRowsRepository.createPickingCandidates(row));
```
afais